nengo_spa/pointer.py

```diff
@@ -109,19 +122,25 @@ def __str__(self):
     def __add__(self, other):
         type_ = infer_types(self, other)
         vocab = None if type_ == TAnyVocab else type_.vocab
-        return SemanticPointer(data=self.v + other.evaluate().v, vocab=vocab)
+        if vocab is None:
+            self._ensure_algebra_match(other)
+        return SemanticPointer(data=self.algebra.superpose(
+            self.v, other.evaluate().v), vocab=vocab)
 
+    @TypeCheckedBinaryOp(Fixed)
     def __radd__(self, other):
-        return self + other
+        type_ = infer_types(self, other)
+        vocab = None if type_ == TAnyVocab else type_.vocab
+        if vocab is None:
+            self._ensure_algebra_match(other)
+        return SemanticPointer(data=self.algebra.superpose(
+            other.evaluate().v, self.v), vocab=vocab)
 
     def __neg__(self):
         return SemanticPointer(data=-self.v, vocab=self.vocab)
 
-    @TypeCheckedBinaryOp(Fixed)
     def __sub__(self, other):
-        type_ = infer_types(self, other)
-        vocab = None if type_ == TAnyVocab else type_.vocab
-        return SemanticPointer(data=self.v - other.evaluate().v, vocab=vocab)
+        return self + (-other)
```

Comment:
Why does `__add__` need type checking, while `__sub__` doesn't?

Reply:
Because sub re-expresses `a - b` as `a + (-b)` thus using the implementation of `__add__` and `__neg__` which already do type checking.